python/ray/tests/test_scheduling.py

```diff
@@ -519,6 +519,26 @@ def f(s, ref):
     wait_for_condition(lambda: not fetches_queued())
 
 
+def test_nested_tasks(shutdown_only):
+    ray.init(num_cpus=1, resources={"worker": 3})
+
+    @ray.remote(num_cpus=1, resources={"worker": 1})
```

Comment:
Can you make the test not use this weird "worker" resource? It's not testing the original issue.

Instead, you can use a counter actor to track max number of instances of each task type.

Reply:
This worker resource guarantees that we have exactly one of each task type running though. Otherwise we're going to go back to relying on timing, which clearly wasn't reliable before.

Reply:
Doesn't the capping guarantee the max number per type regardless of timing?